zebrad/tests/acceptance.rs

```diff
@@ -980,20 +999,21 @@ async fn metrics_endpoint() -> Result<()> {
     assert!(res.status().is_success());
     let body = hyper::body::to_bytes(res).await;
     let (body, mut child) = child.kill_on_error(body)?;
-    assert!(
-        std::str::from_utf8(&body)
-            .expect("metrics response is valid UTF-8")
-            .contains("metrics snapshot"),
-        "metrics exporter returns data in the expected format"
-    );
-
     child.kill()?;
 
     let output = child.wait_with_output()?;
     let output = output.assert_failure()?;
 
+    output.any_output_line_contains(
+        "metrics snapshot",
+        &body,
+        "metrics exporter response",
+        "the metrics response header",
+    )?;
+    std::str::from_utf8(&body).expect("unexpected invalid UTF-8 in metrics exporter response");
```

Comment:
If we do this check using `output`, we get error context from the process, including the process logs.